papers/oleksandr_pavlyk/sklearn.rst

```diff
@@ -0,0 +1,130 @@
+OpenBLAS and MKL uses threaded based parallelism to utilize multicores of modern CPUs.
+Unfortunately  BLAS/LAPACK’s functions are too low level primitives and their usage is often not very efficient comparing to possible high-level parallelism.
+For high-level parallelism scikit-learn uses multiprocessing approach that is not very efficient from technical point of view.
+On the other hand Intel provides Intel |R| Data Analytics Acceleration Library (Intel(R) DAAL) that helps speed up big data analysis by providing highly optimized algorithmic building blocks for all stages of data analytics (preprocessing, transformation, analysis, modeling, validation, and decision making) in batch, online, and distributed processing modes of computation.
+It is originally written in C++ and provides Java and Python bindings.
+In spite of the fact that DAAL is heavily optimized for all Intel Architectures including Xeon Phi it’s very questionable how to use DAAL binding from Python.
+DAAL bindings for python are generated automatically and reflects original C++ API very close that makes its usage quite complicated because of not native for python programmers idioms and chary documentation.
+
+To mix the power of well optimized for HW native code with familiar ML API Intel Python Distribuition started efforts on scikit-learn optimization.
+Therefore, beginning from 2017 U2 IDP includes scikit-learn with daal4sklearn module.
+Specifically, Update 2 optimizes Principal Component Analysis (PCA), Linear and Ridge Regressions, Correlation and Cosine Distances, and K-Means. Speedups may range from 1.5x to 160x.
+
+There are no exact matching between sklearn and DAAL API and they aren’t fully compatible for all inputs so for the cases when daal4sklearn detects incompatibility it fallbacks to original sklearn’s implementation.
+
+Daal4sklearn is enabled by default but provides simple API that allows disabling its functionality:
+
+.. code-block:: python
+
+        from sklearn.daal4sklearn import dispatcher
+        dispatcher.disable()
+        dispatcher.enable()
+
+We prepared several benchmarks to demonstrate performance that can be achieved with DAAL.
+
+.. code-block:: python
+
+        from __future__ import print_function
+        import numpy as np
+        import timeit
+        from numpy.random import rand
+        from sklearn.cluster import KMeans
+
+        import argparse
+        argParser = argparse.ArgumentParser(prog="pairwise_distances.py",
+                                            description="sklearn pairwise_distances benchmark",
+                                            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+        argParser.add_argument('-i', '--iteration', default='10', help="iteration", type=int)
+        argParser.add_argument('-p', '--proc', default=-1, help="n_jobs for algorithm", type=int)
+        args = argParser.parse_args()
```

Comment:
I don't think we need all that code here. Let's publish it on github and provide the link here. Please  leave only vital part of code snippets @dvnagorny